src/twisted/names/test/test_client.py

```diff
@@ -528,6 +528,25 @@ def query(self, address, query, timeout=10, id=None):
         self.assertEqual(len(set(protocols)), 2)
 
 
+    def test_ipv6Resolver(self):
+        """
+        If the resolver is ipv6, open a ipv6 port.
+        """
+        interfaces = []
+
+        # Can't use a plain FakeReactor, because otherwise startListening on
+        # DNSDatagramProtocol will be called.
+        class FakeReactor(test_util.MemoryReactor):
+            def listenUDP(self, port, *args, **kwargs):
+                interfaces.append(kwargs.get('interface'))
+                super(FakeReactor, self).listenUDP(port, *args, **kwargs)
+
+        resolver = client.Resolver(servers=[('::1', 53)],
+                                   reactor=FakeReactor())
+        resolver.query(dns.Query(b'foo.example.com'))
+        self.assertEqual(interfaces, ['::'])
```

Comment:
`MemoryReactor` will give you back the information you need without needing this subclass.  There should be one listening udp port after the query and you can get it from `fake.udpPorts`.  The value is a `MemoryDatagramTransport` with a `getHost` method that describes where it's listening.

So:

`[(proto, transport)]  = fake.udpPorts.items(); interface = transport.getHost().host; assertEqual("::", interface)` - more or less.  `MemoryDatagramTransport` doesn't actually _notice_ it's been told to listen on IPv6, of course, but I don't think this matters in this case (it would be a nice thing to fix when merging `test_util.MemoryReactor` into `proto_helpers.MemoryReactor`).